server/game/cards/plots/07/thewhiteshadows.js

```diff
@@ -0,0 +1,28 @@
+const PlotCard = require('../../../plotcard.js');
+
+class TheWhiteShadows extends PlotCard {
+    setupCardAbilities() {
+        this.reaction({
+            when: {
+                onCharacterKilled: event => this.controller !== event.card.controller
+            },
+            handler: context => {
+                this.controller.putIntoPlay(context.event.card);
+                this.game.addMessage('{0} uses {1} to put {2} into play under their control, blank', 
+                                      this.controller, this, context.event.card);
+
+                this.untilEndOfPhase(ability => ({
+                    match: context.event.card,
+                    effect: [
+                        ability.effects.blank,
+                        ability.effects.moveToDeadPileIfStillInPlay()
```

Comment:
Ah, yes, you are correct in that they should be linked. I think the effect duration should still be `atEndOfPhase` though, instead of `untilEndOfPhase` to match card wording. It reads a bit weird saying 'at end of phase the card is blank' but I believe it should work properly - the card will be blank *until* the 'at end of phase' event fires.